### PR TITLE
Soft remove

### PR DESCRIPTION
**Pull request by markusn**
The SQL queries are very naive, ideas on improvements as well as indexes to be used are welcome.

### CODE REVIEW

File: test/feature/entity-feature.js [hunk 1]
```diff
@@ -42,4 +42,127 @@ Feature("Entity", () => {
       savedEntity.attributes.name.should.equal(entity.attributes.name);
     });
   });
+
+  Scenario("Save, remove and try to load", () => {
```
Comment (lindaandersson) [inline test/feature/entity-feature.js]: Jag tycker att det skulle vara bra om remove testen låg i en egen test fil.

**Comment by markusn** [comment]
@theneubeck removed date field and removed bit field or just a removed date field?

File: test/feature/entity-feature.js (hunk 1 above)
Comment (theneubeck) [reply test/feature/entity-feature.js]: +1

**Comment by theneubeck** [comment]
@markusn, I discussed this with JP. We agreed on this: deleted as boolean, add deleted into the document (if possible). What do you think?

**Comment by markusn** [comment]
Sounds good...

**Pull request merged**